src/software/simulated_tests/simulated_test_simulator.cpp

```diff
@@ -0,0 +1,65 @@
+#include "software/simulated_tests/simulated_test_simulator.h"
+
+#include <pb_decode.h>
+
+#include "software/simulation/simulator.h"
+
+// We use a static variable here to hide simulator.h's NanoPb PrimitiveSetMsg from users
+// of this class
+static std::shared_ptr<Simulator> simulator_static;
```

Comment:
Is this our only option to achieve this?

If so, we should expand on the comment above on why we need to hide the proto primitive definition and why its only applicable to simulated tests, and just a bit more justification defending this approach

Reply:
actually, I can forward Simulator and hold shared_ptr to it